plugins/state_history_plugin/include/eosio/state_history_plugin/session.hpp

```diff
@@ -250,15 +250,24 @@ class session final : public session_base {
                      block_to_send->blocks_result_base.this_block  = {self.current_blocks_request.start_block_num, *this_block_id};
                      if(const std::optional<chain::block_id_type> last_block_id = self.get_block_id(self.next_block_cursor - 1))
                         block_to_send->blocks_result_base.prev_block = {self.next_block_cursor - 1, *last_block_id};
-                     if(chain::signed_block_ptr sbp = get_block(self.next_block_cursor); sbp && self.current_blocks_request.fetch_block)
-                         block_to_send->blocks_result_base.block = fc::raw::pack(*sbp);
+                     if (self.current_blocks_request.fetch_block) {
+                        if (chain::signed_block_ptr sbp = get_block(*this_block_id)) {
+                           block_to_send->blocks_result_base.block = fc::raw::pack(*sbp);
+                        } else {
+                           fc_wlog(logger, "Unable to retrieve block ${bn} : ${id}",
+                                   ("bn", chain::block_header::num_from_id(*this_block_id))("id", *this_block_id));
+                        }
+                     }
                      if(self.current_blocks_request.fetch_traces && self.trace_log)
                         block_to_send->trace_entry = self.trace_log->get_entry(self.next_block_cursor);
                      if(self.current_blocks_request.fetch_deltas && self.chain_state_log)
                         block_to_send->state_entry = self.chain_state_log->get_entry(self.next_block_cursor);
                      if(block_to_send->is_v1_request && *self.current_blocks_request_v1_finality && self.finality_data_log)
                         block_to_send->finality_entry = self.finality_data_log->get_entry(self.next_block_cursor);
+                  } else {
+                     fc_wlog(logger, "Unable to retrieve block id for block ${bn}", ("bn", self.next_block_cursor));
```

Comment:
This log is certainly more reasonable than the one above since in this case the empty messages that will be sent to the client are kinda mysterious since everything but head & lib is empty, so the log helps out in adding context.

Since I believe the only way we realistically expect to encounter this situation is when a client requests blocks below the `min(blog, trace, state, finality)`, we could consider,

* easy: adding  `remote_endpoint_string` to the message; this alone will help convey the warning is the result of a client's request and not something internally wrong
* medium: possibly only printing the log once per a request to avoid massive log spam if someone mistakenly requests a large range they don't have; like when starting fresh from snapshot but requesting from block 1
* hard: possibly printing a more helpful log such as "client ${remote_endpoint_string} requested blocks from ${bn} but minimum block with any results is ${min_block}; empty results will be streamed until minimum block"; of course `min_block` will require a new function to compute `min(blog, trace, state, finality)`

imo we don't need to do any of these for now; but the 'easy' one might be nice

Reply:
I created the following, but realized that all this (except `earliest_available_block_num()`) is available via the SHiP status request. I think we should just remove the log output and let the user use status request if they do not get what they anticipate. 

![image](https://github.com/user-attachments/assets/31a23c7b-9e4a-4140-9d7d-0c1941883d20)